cmd/zed/agents/zfs_retire.c

```diff
@@ -318,6 +329,32 @@ zfs_retire_recv(fmd_hdl_t *hdl, fmd_event_t *ep, nvlist_t *nvl,
 		    &vdev)) == NULL)
 			return;
 
+		/* Can't replace a spare with another one: offline the device */
+		if (nvlist_lookup_string(nvl, FM_EREPORT_PAYLOAD_ZFS_VDEV_TYPE,
+		    &type) == 0 && strcmp(type, VDEV_TYPE_SPARE) == 0) {
+			char *dev_name;
+
+			dev_name = zpool_vdev_name(NULL, zhp, vdev, B_FALSE);
+			fmd_hdl_debug(hdl, "zpool_vdev_offline '%s'", dev_name);
+			zpool_vdev_offline(zhp, dev_name, B_TRUE);
```

Comment:
Note: this doesn't currently work (offlining a spare). Once resolved we could probably close https://github.com/zfsonlinux/zfs/issues/1105.